nifi-extension-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/stream/pause/StandardRecordProcessorBlocker.java

```diff
@@ -17,19 +17,57 @@
 package org.apache.nifi.processors.aws.kinesis.stream.pause;
 
 import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
 
+/**
+ * Standard implementation of {@link RecordProcessorBlocker} that allows blocking calls to {@link #await()} by:
+ * <ul>
+ *     <li>explicit call to {@link #block()}</li>
+ *     <li>timeout since last call to {@link #unblock()}</li>
+ * </ul>
+ * Unblocking is done by calling {@link #unblock()} or {@link #unblockAndDisableTimeout()}. The latter is used to disable the timeout mechanism to allow shutdown of background threads that could
+ * otherwise block indefinitely due to no calls to {@link #unblock()}.
+ */
 public class StandardRecordProcessorBlocker implements RecordProcessorBlocker {
 
     private CountDownLatch blocker = new CountDownLatch(0);
 
+    static final long BLOCK_AFTER_TIMEOUT_MILLIS = TimeUnit.SECONDS.toMillis(2);
+    private final Supplier<Long> timestampProvider;
+    private long timeoutAnchor = 0L;
```

Comment:
The naming of this property is unclear, it should be renamed to something that more clearly describes its purpose.

Reply:
Renamed to `timeoutBase`